pkg/api/datasources.go

```diff
@@ -129,8 +130,16 @@ func DeleteDataSourceByName(c *models.ReqContext) Response {
 	return Success("Data source deleted")
 }
 
+// reURL is a regexp to detect if a URL specifies the protocol. We match also strings where the actual protocol is missing
+// (i.e., "://"), in order to catch these as invalid when parsing.
+var reURL = regexp.MustCompile("^[^:]*://")
+
 func validateURL(u string) Response {
 	if u != "" {
+		// Make sure the URL starts with a protocol specifier, so parsing is unambiguous
+		if !reURL.MatchString(u) {
```

Comment:
It would be nice with some tests for this. I always find it easier to understand regexp patterns if I can read tests :)

Reply:
@bergquist Gotcha, I will add a test or two.